File: 853. Car Fleet.go
package main

import (
	"fmt"
	"sort"
)

type st struct {
	pos   int
	speed int
	time  float32
}

func carFleet(target int, position []int, speed []int) int {
	a := make([]st, 0)
	for i, _ := range position {
		ti := float32(target-position[i]) / float32(speed[i])
		t := st{pos: position[i], speed: speed[i], time: ti}
		a = append(a, t)
	}

	sort.Slice(a, func(i, j int) bool {
		return a[i].pos > a[j].pos
	})

	fmt.Print(a)

	res := make([]st, 0)
	for _, v := range a {
		if len(res) == 0 {
			res = append(res, v)
		} else {
			if v.time > res[len(res)-1].time {
				res = append(res, v)
			}
		}
	}

	return len(res)
}

func main() {
	target := 12
	position := []int{10, 8, 0, 5, 3}
	speed := []int{2, 4, 1, 1, 3}
	fmt.Println(carFleet(target, position, speed))
}

File: 853. Car Fleet.py

def carFleet(target, position, speed):
    """
    :type target: int
    :type position: List[int]
    :type speed: List[int]
    :rtype: int
    """
    a = [{'s':speed[i], 'p':position[i], 't': ((target - position[i])/float(speed[i]))} for i in range(len(position)) ]
    a = sorted(a, key= lambda x: x['p'], reverse = True)
    # t = [(target - position[i])/speed[i] for i in range(len(position))]
    st = []
    for i in a:
        if len(st) == 0:
            st.append(i)
        else:
            if i['t'] > st[-1]['t']:
                st.append(i)
    return len(st)

target = 12
position = [10,8,0,5,3]
speed = [2,4,1,1,3]

print (carFleet(target, position, speed))